wgpu-core/src/validation.rs

```diff
@@ -346,7 +349,7 @@ impl Resource {
                             },
                             wgt::TextureSampleType::Depth => naga::ImageClass::Depth,
                         };
-                        (class, GlobalUse::READ)
+                        (class, GlobalUse::READ | GlobalUse::QUERY)
```

Comment:
let's also enable this for uniform buffers, read-only storage buffers, and all of the storage textures